### ISSUE TITLE
Add possibility to delete photos when using external image viewer 

### ISSUE THREAD

**Issue by Discostu36** [MarcusWolschon/osmeditor4android#1913]
At the moment, it is not possible to delete photos inside of the app when using external photo viewer. I need to switch to internal photo viewer for that. I think this should be changed, if possible.

**Comment by simonpoole** [comment]
Well the problem is that such a mechanism would require using the internal photo viewer (there needs to be some way to select and display the photo you want to delete).

**Comment by Discostu36** [comment]
Is it possible to give the external viewer permission to delete?

**Comment by simonpoole** [comment]
> Is it possible to give the external viewer permission to delete?

I don't think so (IMHO the only thing you can grant is persistent read access), but I'll check.

**Comment by Discostu36** [comment]
![Screenshot_20221029-164448.png](https://user-images.githubusercontent.com/30999326/198838001-c01a2431-3849-431c-af12-f6128c3fc356.png)

This happens when I try to edit an image shared by Vespucci. I guess to make deletion possible you'd need to pass the image from its file system path (Downloads/Vespucci) to the image viewer as it might not be possible for it to delete a file only knowing this (private?) file path.

(Keep in mind that I have virtually no experience in Android programming yet, so this is only guesswork 😀)

**Comment by simonpoole** [comment]
Conventional *ix file operations don't work on Android outside of directories and files the app has created itself. File access happens via Uri that are specific to ContentProviders. In any case I'll check if granting FLAG_GRANT_WRITE_URI_PERMISSION will allow a file to be deleted.

**Comment by simonpoole** [comment]
A further note after some closer inspection in your example I don't see how that could work as the image in question is indexed via the mediastore and isn't owned by the app to start with.

**Comment by Discostu36** [comment]
Two thoughts about that:
1) Yes, the example was not well-chosen. This should primarily work for photos that were taken with Vespucci
2) Can you explain to me why Vespucci needs to “own” the file for this to work? My file manager app also doesn't own the photos. But if I open a photo from it, it is opened in Gallery with all editing rights.

**Comment by simonpoole** [comment]
Since Android 11 there is a special permission for file managers and similar  https://developer.android.com/about/versions/11/privacy/storage#all-files-access We could in theory naturally ask for this too, but would have to live with even more issues with google if they would even allow us to have it to start with.

**Comment by simonpoole** [comment]
Revisiting this, it would seem that to allow external image viewers to delete images we would need to store them in the normal locations and index them via the media store. This is likely not an issue (it will need a preference), but has the downside that there is then no obvious way to keep vespucci originating images separate.
